src/hotspot/share/logging/logAsyncWriter.hpp

```diff
@@ -213,6 +214,11 @@ class AsyncLogWriter : public NonJavaThread {
 
   static AsyncLogWriter* instance();
   static void initialize();
+  // Returns true if initialized, otherwise false
+  static bool enqueue_if_initialized(LogFileStreamOutput& output, const LogDecorations& decorations,
+                                     const char* msg);
+  static bool enqueue_if_initialized(LogFileStreamOutput& output,
+                                     LogMessageBuffer::Iterator msg_iterator);
```

Comment:
Are these really part of the public API? Couldn't the additional logic just be folded into the enqueue method?

Reply:
Hm, yeah, we can fold it into the `enqueue` methods since they basically do nothing as is.